playkit/src/main/java/com/kaltura/playkit/player/ExoPlayerWrapper.java

```diff
@@ -406,9 +406,13 @@ private MediaSource buildExternalSubtitleSource(int subtitleId, PKExternalSubtit
                     .connectTimeout(DefaultHttpDataSource.DEFAULT_CONNECT_TIMEOUT_MILLIS, TimeUnit.MILLISECONDS)
                     .readTimeout(DefaultHttpDataSource.DEFAULT_READ_TIMEOUT_MILLIS, TimeUnit.MILLISECONDS);
 
-            final okhttp3.EventListener.Factory okListenerFactory = profiler.getOkListenerFactory();
-            if (okListenerFactory != null) {
-                builder.eventListenerFactory(okListenerFactory);
+            if (profiler != Profiler.NOOP) {
+                final okhttp3.EventListener.Factory okListenerFactory = profiler.getOkListenerFactory();
+                if (okListenerFactory != null) {
+                    builder.eventListenerFactory(okListenerFactory);
+                }
+            } else {
+                builder.eventListener(analyticsAggregator);
```

Comment:
Please check if there can be more than one listener -- then we will be able to have both profiler and analytics aggregator.

Reply:
move the  builder.eventListener(analyticsAggregator); to be first without else